src/main/scala/com/databricks/spark/sql/perf/mllib/feature/StringIndexer.scala

```diff
@@ -0,0 +1,31 @@
+package com.databricks.spark.sql.perf.mllib.feature
+
+import org.apache.spark.ml
+import org.apache.spark.ml.linalg.Vector
+import org.apache.spark.ml.PipelineStage
+import org.apache.spark.sql._
+
+import com.databricks.spark.sql.perf.mllib.OptionImplicits._
+import com.databricks.spark.sql.perf.mllib.data.DataGenerator
+import com.databricks.spark.sql.perf.mllib.{BenchmarkAlgorithm, MLBenchContext, TestFromTraining}
+
+/** Object for testing StringIndexer performance */
+object StringIndexer extends BenchmarkAlgorithm with TestFromTraining with UnaryTransformer {
+
+  override def trainingDataSet(ctx: MLBenchContext): DataFrame = {
+    import ctx.params._
+    import ctx.sqlContext.implicits._
+
+    DataGenerator.generateOverlappedString(ctx.sqlContext,
+      numExamples, ctx.seed(), numPartitions, ldaNumVocabulary).select($"label".as(inputCol))
```

Comment:
Nit: Instead of renaming the string column from "label" to `inputCol` here (and everywhere else the `generateOverlappedString` will be called), maybe pass in the desired name of the string column as a parameter to `generateOverlappedString`.